app/api/api.rb

```diff
@@ -19,15 +21,29 @@ def is_public_request?
       request.path.include?('/api/v1/public/')
     end
 
-    def group_by_molecule(samples)
+    def group_by_molecule(samples,own_collection = false)
       groups = Hash.new
-      samples.each do |s|
-        moleculeName = get_molecule_name(s)
-        serialized_sample = ElementPermissionProxy.new(current_user, s).serialized
-        if !groups[moleculeName]
-          groups[moleculeName] = [].push(serialized_sample)
-        else
-          groups[moleculeName] = groups[moleculeName].push(serialized_sample)
+
+      if own_collection
+        samples.each do |s|
+          SampleSerializer::Level10.new(s, 10).serializable_hash
+          moleculeName = get_molecule_name(s)
+          serialized_sample = SampleSerializer::Level10.new(s, 10).serializable_hash
```

Comment:
Since other lines is the same, can we just apply the if on this line?
`serialized_sample = own_collection ? SampleSerializer::Level10.new(s, 10).serializable_hash : ElementPermissionProxy.new(@current_user, s, @user_ids).serialized`